scripts/osfstorage/migrate_to_generic.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import logging
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from website.app import init_app
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+from website.files import models
+from website.addons.osfstorage import model as osfstorage_model
+
+logger = logging.getLogger(__name__)
+
+
+def do_migration():
+    logger.info('Migration: OsfStorageFileNode -> FileNode')
+    migrate_filenodes()
+    logger.info('Migration: OsfStorageTrashedFileNode -> TrashedFileNode')
+    migrate_trashedfilenodes()
+
+    logger.info('Checking that all File versions have been migrated...')
+    diff = osfstorage_model.OsfStorageFileVersion.find().count() - models.FileVersion.find().count()
+    if diff != 0:
+        logger.error('{} OsfStorageFileVersions did not get migrated'.format(diff))
+        logger.error('This is most likely because they are orphaned')
+        logger.error('This is not a show stopper; The migration was still successful')
+    else:
+        logger.info('Migration successful')
+
+
+def migrate_trashedfilenodes():
+    for trashed in osfstorage_model.OsfStorageTrashedFileNode.find():
+        logger.debug('Migrating OsfStorageTrashedFileNode {}'.format(trashed._id))
+
+        parent = osfstorage_model.OsfStorageTrashedFileNode.load(trashed.to_storage()['parent'])
+        if parent is None:
+            parent = osfstorage_model.OsfStorageFileNode.load(trashed.to_storage()['parent'])
+
+        if trashed.node_settings is None:
+            logger.warning('OsfStorageTrashedFileNode {} has no node_settings; skipping'.format(trashed._id))
+            continue
+
+        models.TrashedFileNode(
+            _id=trashed._id,
+            versions=[translate_versions(trashed.versions)],
+            node=trashed.node_settings.owner,
+            parent=parent,
+            is_file=trashed.kind == 'file',
+            provider='osfstorage',
+            name=trashed.name,
+            path='/' + trashed._id + ('' if trashed.kind == 'file' else '/'),
+            materialized_path=''
+        ).save()
+
+
+def migrate_filenodes():
+    for node_settings in osfstorage_model.OsfStorageNodeSettings.find():
+        if node_settings.owner is None:
+            logger.warning('OsfStorageNodeSettings {} has no parent; skipping'.format(node_settings._id))
+            continue
+        logger.info('Migrating files for {}'.format(node_settings.owner.title))
+        root_node = osfstorage_model.OsfStorageFileNode.load(node_settings.to_storage()['root_node'])
+        if root_node is None:
+            logger.warning('OsfStorageNodeSettings {} has no root_node; skipping'.format(node_settings._id))
+            continue
+        list(osfstorage_model.OsfStorageNodeSettings.find(Q('node_settings', 'eq', node_settings._id)))
```

Comment:
It seems incorrect; you're querying the `node_settings` field of the `OsfStorageNodeSettings` model.